### DISCUSSION TITLE
Some ideas for future versions...

### DISCUSSION THREAD

**Discussion by jmaraujouy** [SubtitleEdit/subtitleedit#7349] [Ideas]
SE keeps getting better and better. Here are some ideas that (maybe) could be implemented in future versions...

* <strike>Choosing between language-based interjection lists.</strike> ✅ **COMPLETED** in v4.0.1 BETA 82 _(Thanks, Nik!)_
* <strike>The possibility to disable unused languages.</strike> ✅ **COMPLETED** in v4.0.1 BETA 82 _(Thanks, Nik!)_
* The possibility to translate plugins names _(fig.1)_
* Nesting groups in the 'Multiple Replace' window. _(fig.2)_
* Converting HI text that makes a reference to "Both" & "All" speakers in a single line, to a dialog format. _(fig.3)_
* Checking for BETA version updates. _(fig.4)_
* Use the keyboard to modify the _selected subtitle_ sync & order in the list view.

Figure 1: Localization of plugin names
![imagen](https://github.com/SubtitleEdit/subtitleedit/assets/117078256/296b233b-a16e-4415-9b00-ca99f9fb731a)

Figure 2: Nested groups in 'Multiple Replace' window
![imagen](https://github.com/SubtitleEdit/subtitleedit/assets/117078256/fe6966a7-51fe-48c1-a1d0-5db96221b445)

Figure 3: Convert HI 'Both' & 'All' to dialog format
![imagen](https://github.com/SubtitleEdit/subtitleedit/assets/117078256/190eeacd-7340-49b7-936d-251bb7785b02)

Figure 4: Check for BETA updates
![imagen](https://github.com/SubtitleEdit/subtitleedit/assets/117078256/c76011bd-e518-4b94-b806-a730e98193be)

**Comment by jmaraujouy** [comment]
I think, at least, the option to choose between language-based interjection lists might be very useful.

Because what constitutes an interjection in one language may be a valid word in another language. In my case, it happens often with the interjection **"_Ha_"** in english, and the verb **"_Ha_"** in spanish, which means "has".

An example: "El sol ha salido," translates "The sun has risen".

So I have to keep adding and deleting that word from the list, depending on whether the subtitle is in Spanish or English. This is not the only case, but it's the more substantial I can think of right now.

<br /><hr /><br />

Here's another idea: Adding an option to the "Remove text for hearing impaired" so HI text that makes a reference to "Both" and/or "All" speakers in a single line, could be converted to a dialog format. Here's an example:

Convert this:
```
139
00:10:27,461 --> 00:00:28,429
Both: Hi!
```

To this:
```
139
00:10:27,461 --> 00:00:28,429
- Hi!
- Hi!
```

**Comment by jmaraujouy** [comment]
Here's another small idea that might be useful: Checking for BETA version updates.

![imagen](https://github.com/SubtitleEdit/subtitleedit/assets/117078256/ffb1c62f-2a1f-4e66-b25f-15924b2a9b65)

**Comment by niksedk** [comment]
I've added language specific interjections here (has also been on my todo list for years):
![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/b38d38f6-b1a2-4248-9fde-0bb1c74c5b4a)

Multiple people have asked for "language filter".
Now in Options - Settings - General:
![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/a948a0c6-76d1-4aa7-b6b5-93a1fa111a60)

And also in language drop downs (hopefully most):
![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/d403e652-082b-4f13-880c-a32ecf7d0a3f)



Latest beta link: https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.1/SubtitleEditBeta.zip

Let me know how it works.

**Comment by LeonCheung** [comment]
Talking about the 'Fix Common Errors' window, maybe #5476 could be discussed a bit here? I mean the **Search** tool and **Selection Set**. Search tool is for quick toggling/untoggling. Selection Set is for quick switch among different sets, because different projects may need different checks, say, some project are very length-sensitive, while others are not. And, when working with ASSA, overlaps doesn't matter much. etc, etc, etc... Change toggling sets one by one, every time, and 100% manually, would be really uninfficient...

Btw, Surprisingly, any toggles there will be rememberd instantly even I click Cancel... Say, last time I clicked Select All, Invert selection, and Select default (curiosity kills the cat...), my previous setup was gone and I have no way to recover but having to toggle them one by one. A little sad story to share. 😂

**Comment by jmaraujouy** [comment]
Fantastic job, @niksedk !

I really appreciate the time and effort you put to make SE better and better with each update!

I'll upload a `es_interjections.xml` list soon.

**Comment by jmaraujouy** [comment]
> Let me know how it works.

@niksedk ,

I'm not sure if I'm doing something wrong, but I think is not working with language based lists.

Here is an example. I have an English list that contains the interjection 'Huh', and a Spanish list that **does not** include that word. When I change languages, 'Huh' is still showing as an interjection, even when it is supposedly not in the list.


https://github.com/SubtitleEdit/subtitleedit/assets/117078256/c3dede08-7999-4d19-98e5-1a372e5166dd

**Reply by jmaraujouy** [reply]
**UPDATE:** It works if I click on _Edit_, and then _OK_.  If I change the language back, I must do it again: Edit -> Ok.

Here's a video:

https://github.com/SubtitleEdit/subtitleedit/assets/117078256/dfd03327-1b0f-4446-beab-a04c4169c19f

**Reply by niksedk** [reply]
Thx for testing - hopefully better in latest beta: https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.1/SubtitleEditBeta.zip

**Reply by jmaraujouy** [reply]
Yes, seems to be fixed and I can change interjections lists in the latest beta (see video). 👍🏻

One tiny detail: Is it possible to make the list to auto refresh when you change languages?

https://github.com/SubtitleEdit/subtitleedit/assets/117078256/0fc40fbe-06c4-4d1a-b059-539e7f19d79c

**Reply by niksedk** [reply]
thx - should auto refresh now: https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.1/SubtitleEditBeta.zip

**Reply by jmaraujouy** [reply]
Auto refresh doesn't seems to be working. 4.0.1 NEXT Beta v108)

**Reply by niksedk** [reply]
thx for testing... beta updated: https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.1/SubtitleEditBeta.zip

**Reply by jmaraujouy** [reply]
@niksedk It is fixed. 👍🏻 Subtitles list auto refreshes when changing languages.

Thank you, Nikolaj!

**Comment by sasamavros** [comment]
It would be great to be able to keep our customized shortcuts in subtitle edit

**Reply by darnn** [reply]
To do this, copy the settings.xml file from the folder of the previous version of Subtitle Edit you were using.

**Reply by sasamavros** [reply]
Thanks for replying, but I don't see it in my Windows/Subtitle Edit file. What am I doing wrong?

**Reply by jmaraujouy** [reply]
> To do this, copy the settings.xml file from the folder of the previous version of Subtitle Edit you were using.

I'm not 100% sure, but I think if you put your settings file in the User Profile folder, and not in your Program Files folder, it won't be overwritten when the program gets an update.

User profile folder path:
`c:\Users\{Your Username}\AppData\Roaming\Subtitle Edit`

Program files folder:
`c:\Program Files\Subtitle Edit`

Adjust "c:" to your assigned drive letter (if not "c:")
